xarray/tests/test_groupby.py

```diff
@@ -3161,6 +3161,81 @@ def test_season_to_month_tuple(self):
             (10, 11),
         )
 
+    def test_season_grouper_raises_error_if_months_are_not_valid_or_not_continuous(
+        self,
+    ):
+        calendar = "standard"
+        time = date_range("2001-01-01", "2002-12-30", freq="D", calendar=calendar)
+        da = DataArray(np.ones(time.size), dims="time", coords={"time": time})
+
+        with pytest.raises(KeyError, match="IN"):
+            da.resample(time=SeasonResampler(["INVALID_SEASON"]))
+
+        with pytest.raises(KeyError, match="MD"):
+            da.resample(time=SeasonResampler(["MDF"]))
+
+    @pytest.mark.parametrize("calendar", _ALL_CALENDARS)
+    def test_season_grouper_with_seasons_spanning_calendar_year_uses_previous_year(
+        self, calendar
+    ):
```

Comment:
This test currently fails with the expected result. I put two expected results, one with previous year "ND" (fails) and the other with the same year "ND" (passes, commented out).